cmd/join.go

```diff
@@ -256,11 +262,18 @@ func setupAgent(serverIP, ip net.IP, port int, user, sshKeyPath, joinToken, k3sE
 	defer operator.Close()
 
 	installStr := createVersionStr(k3sVersion, k3sChannel)
-	installAgentCommand := fmt.Sprintf("curl -sfL https://get.k3s.io/ | K3S_URL='https://%s:6443' K3S_TOKEN='%s' %s sh -s - %s",
+
+	serverAgent := false
+
+	installK3sExec := makeJoinExec(
 		serverIP.String(),
 		strings.TrimSpace(joinToken),
 		installStr,
-		k3sExtraArgs)
+		k3sExtraArgs,
+		serverAgent,
+	)
+
+	installAgentCommand := fmt.Sprintf("curl -sfL https://get.k3s.io | %s", installK3sExec)
```

Comment:
I wonder if we have `curl -sfL https://get.k3s.io` repeated three times now for the agent, agent-server and server install?